Realm/Realm.UnityWeaver/UnityWeaver.cs

```diff
@@ -216,6 +216,12 @@ private static bool WeaveAssemblyCore(string assemblyPath, IEnumerable<string> r
 
                     var results = weaver.Execute(analyticsConfig);
 
+                    if (results.ErrorMessage != null)
```

Comment:
Would it be an option to let the `weaver.Execute` throw to make sure we don't forget that check in other occasions?
The `catch` seems to log the same result then anyway.

Reply:
That would be a bad idea for two reasons:
1. Those are different error cases - the one where we have an error message is expected and likely an issue with the user's code. The one being caught is unexpected and likely an issue with our code. We can (and perhaps should) modify the messaging in the `Exception` case to instruct the user to open an issue if they encounter it.
2. The weaver is being used by Fody and there we don't have a `catch` clause - we rely on Fody to handle exceptions in this case and there's a notable difference between returning an error message and throwing an exception.

Reply:
Good idea. The modified message would clarify it for the user (and for me 🙃 ).